### ISSUE TITLE
East Riding of Yorkshire Council Broken

### ISSUE THREAD

**Issue by AlistairHA** [mampfes/hacs_waste_collection_schedule#1115]
Error generated in HA when using East Riding of Yorkshire

**ERROR (SyncWorker_6) [waste_collection_schedule.source_shell] fetch failed for source East Riding of Yorkshire Council:
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/eastriding_gov_uk.py", line 37, in fetch
    api_key = re.findall(REGEX["API_KEY"], r1.text)[0]
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range**


Double checked Configured everything correctly, tried with several postcodes/uprn.
Also tried with an alternate council feed = OK.
API key expired?

UPDATE:
The url  'https://www.eastriding.gov.uk/environment/bins-rubbish-recycling/bins-and-collections/bin-collection-dates/' in teh section  # get api_key and licensee    is returning <Response [403]>

**Comment by dt215git** [comment]
Looks like they've changed the website and the api key is no long in plaintext on the page. The regex that tries to extract it isn't finding anything hence the _list index out of range_ error. The subsequent query that returns the collection schedule doesn't look any different, so it's a case of trying to find out how it's now getting the api key.

**Comment by 5ila5** [comment]
fixed by #1119 will be part of release 1.42.0 or is available now as master/git version (up to date clone of this repository)
